src/fileformat/file_format/pe/pe_format.cpp

```diff
@@ -598,6 +601,657 @@ void PeFormat::loadRichHeader()
 	richHeader->setBytes(header.getDecryptedHeaderBytes());
 }
 
+/**
+ * Load visual basic header
+ */
+void PeFormat::loadVisualBasicHeader()
+{
+	auto allBytes = getBytes();
+	std::vector<std::uint8_t> bytes;
+	unsigned long long version = 0;
+	unsigned long long vbHeaderAddress = 0;
+	unsigned long long vbHeaderOffset = 0;
+	unsigned long long vbProjectInfoOffset = 0;
+	unsigned long long vbComDataRegistrationOffset = 0;
+	std::string projLanguageDLL;
+	std::string projBackupLanguageDLL;
+	std::string projExeName;
+	std::string projDesc;
+	std::string helpFile;
+	std::string projName;
+	std::size_t offset = 0;
+	struct VBHeader vbh;
+
+	if (!isVisualBasic(version))
+	{
+		return;
+	}
+
+	// first instruction is expected to be PUSH <vbHeaderAddress> (0x68 <b0> <b1> <b2> <b3>)
+	if (!getEpBytes(bytes, 5) || bytes.size() != 5 || bytes[0] != 0x68)
+	{
+		return;
+	}
+
+	vbHeaderAddress = bytes[4] << 24 | bytes[3] << 16 | bytes[2] << 8 | bytes[1];
+	if (!getOffsetFromAddress(vbHeaderOffset, vbHeaderAddress))
+	{
+		return;
+	}
+
+	if (!getBytes(bytes, vbHeaderOffset, vbh.headerSize()) || bytes.size() != vbh.headerSize())
+	{
+		return;
+	}
+
+	vbh.signature = *reinterpret_cast<std::uint32_t *>(&bytes.data()[offset]); offset += sizeof(vbh.signature);
```

Comment:
This line is recurring in your changes. I was thinking whether it would be possible for you to either create a function that would do this or even better, move `DynamicBuffer` from `src/unpacker` to `utils` and use it. It basically does what you do but it abstracts away all the casting and also maintains endianess, so you don't have to do byte swaps afterwards.